### DISCUSSION TITLE
Minecraft won't launch, new to making mods (vscode)

### DISCUSSION THREAD

**Discussion by FabledSol** [FabricMC/fabric-api#1753] [Mod Dev Support] [answered]
Hi, so I have been trying to make a mod in fabric and haven't had any crash issues until I tried to make an entity, I looked over the crash and it looks like it has something to do with the renderer. Although I am unsure as to what the issue is. Here is the crash report:

---- Minecraft Crash Report ----
// Who set us up the TNT?

Time: 9/28/21, 1:14 PM
Description: Initializing game

java.lang.IllegalStateException: No renderer registered for revo:cube
	at net.minecraft.client.render.entity.EntityRenderDispatcher.<init>(EntityRenderDispatcher.java:204)
	at net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:542)
	at net.minecraft.client.main.Main.main(Main.java:149)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.client.render.entity.EntityRenderDispatcher.<init>(EntityRenderDispatcher.java:204)
	at net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:542)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:149)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)

-- System Details --
Details:
	Minecraft Version: 1.16.5
	Minecraft Version ID: 1.16.5
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 11.0.12, Eclipse Foundation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Eclipse Foundation
	Memory: 1149974016 bytes (1096 MB) / 1448083456 bytes (1381 MB) up to 3193962496 bytes (3046 MB)
	CPUs: 4
	JVM Flags: 3 total; -Xdebug -Xnoagent -Xrunjdwp:transport=dt_socket,address=localhost:58800,server=n,suspend=y
	Fabric Mods: 
		fabric: Fabric API 0.37.2+1.16
		fabric-api-base: Fabric API Base 0.3.0+c88702897d
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.2.0+6aebe11a7d
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.10+3661eaa17d
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.5+ca58154a7d
		fabric-command-api-v1: Fabric Command API (v1) 1.1.2+95b14da07d
		fabric-commands-v0: Fabric Commands (v0) 0.2.2+ca58154a7d
		fabric-containers-v0: Fabric Containers (v0) 0.1.11+9354966b7d
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.4+5badb62d7d
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.3+ca58154a7d
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.6+61336ba97d
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.1.0+c88702897d
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.3+f941baf07d
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.1+ca58154a7d
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.6+ca58154a7d
		fabric-item-api-v1: Fabric Item API (v1) 1.2.1+ca58154a7d
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.7+2868a2287d
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.4+9354966b7d
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.1+ca58154a7d
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.2.1+ca58154a7d
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.2+ca58154a7d
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.3+ca58154a7d
		fabric-models-v0: Fabric Models (v0) 0.3.0+bc7a746f7d
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.4+5badb62d7d
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.8+ca58154a7d
		fabric-networking-v0: Fabric Networking (v0) 0.3.2+ca58154a7d
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.9.5+c88702897d
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.2+ca58154a7d
		fabric-particles-v1: Fabric Particles (v1) 0.2.4+ca58154a7d
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.4+ca58154a7d
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.4+f3e8589e7d
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.4+ca58154a7d
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.3.0+2868a2287d
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.5+ca58154a7d
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.13+ca58154a7d
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.2+ca58154a7d
		fabric-rendering-v1: Fabric Rendering (v1) 1.6.0+2868a2287d
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.7+6afaa13c7d
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.0+c045166c7d
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.5+9354966b7d
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.11+ca0ef93d7d
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.1+ca58154a7d
		fabric-textures-v0: Fabric Textures (v0) 1.0.6+ca58154a7d
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.7+8183d3957d
		fabric-transfer-api-v1: Fabric Transfer API (v1) 1.0.0+5f6b94517d
		fabricloader: Fabric Loader 0.11.6
		java: OpenJDK 64-Bit Server VM 11
		minecraft: Minecraft 1.16.5
		revo: Reincarnated Evolution ${version}
	Launched Version: Fabric
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: Intel(R) HD Graphics 530 GL version 4.5.0 - Build 25.20.100.6446, Intel
	GL Caps: Using framebuffer using OpenGL 3.0
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: 4x Intel(R) Core(TM) i5-6400 CPU @ 2.70GHz

Any help on this would be appreciated.

**Comment by Juuxel** [comment, marked as answer]
> No renderer registered for revo:cube

This means you need to use Fabric API's [`EntityRendererRegistry`](https://github.com/FabricMC/fabric/blob/1.17/fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/EntityRendererRegistry.java) to add a renderer for your entity. The wiki also has an article about this: https://fabricmc.net/wiki/tutorial:entity

**Reply by FabledSol** [reply]
I had followed a similar tutorial from this video which results in no visual errors or the red lines but failed in the launch
https://www.youtube.com/watch?v=IPUW8odk3yY

Now when I followed the fabric tutorial from the actual website it instead gives me errors around the EntityModelLayer and the EntityRendererFactory any idea on what the issue could be there? When I do quick fix it wants me to create classes for all of the errors but the tutorials didn't mention doing that

Client
![client](https://user-images.githubusercontent.com/77421383/135178354-39c8eed4-412b-46e0-9f9b-85392a92d481.PNG)

Renderer
![renderer](https://user-images.githubusercontent.com/77421383/135178359-95ec12fe-9605-445e-8c10-e1309c97004b.PNG)

.

**Reply by Juuxel** [reply]
Oh, you're still using 1.16. Mojang refactored entity renderers in 1.17 (as the often do...), so the [1.16 revision of the wiki tutorial](https://fabricmc.net/wiki/tutorial:entity?rev=1611967124) is probably more helpful.

**Reply by FabledSol** [reply]
Thanks
